addons/sale/models/sale.py

```diff
@@ -472,22 +472,27 @@ def _prepare_invoice(self):
             .default_get(['journal_id'])['journal_id'])
         if not journal_id:
             raise UserError(_('Please define an accounting sales journal for this company.'))
-        vinvoice = self.env['account.invoice'].new({'partner_id': self.partner_invoice_id.id, 'type': 'out_invoice'})
+        AccountInvoice = self.env['account.invoice']
+        vals = AccountInvoice.default_get(AccountInvoice._fields.keys())
+        vals.update({
+            'partner_id': self.partner_invoice_id.id,
+            'type': 'out_invoice',
+            'company_id': company_id,
+        })
+        vinvoice = AccountInvoice.new(vals)
```

Comment:
```suggestion
        vinvoice = self.env['account.invoice'].new({
            'partner_id': self.partner_invoice_id.id,
            'type': 'out_invoice',
            'company_id': company_id})
```
Updating all the default values may have unknown side-effects and be risky for a stable branch, but I can accept the `company_id` value as a targeted fix for multi-company purposes.